### ISSUE TITLE
"No Result" Option

### ISSUE THREAD

**Issue by dnorthrupva** [pawelczak/EasyAutocomplete#251]
I've seen numerous requests for this but all pretty dated. Has anyone come up with a way to do 'No Results'? I tried using the `onHideListEvent` event to do it but didn't come up with the same results as I expected.

**Comment by milpas999** [comment]
I am Facing the same issue.
I tried one alternate, If there is no data found corresponding to the search term, then i am sending a "No Data found" in response from the server. and it is getting displayed over here.
When user clicks out side anywhere in the page it goes of.

**Comment by mkrsII** [comment]
Just ran into this problem... do something like this:
you need to check if there's nothing in the data source probably first

onHideListEvent

var containerList = $('#AutoCompleteID').next('.easy-autocomplete-container').find('ul'); if ($(containerList).children('li').length <= 0) { $(containerList).html('<li>No results found</li>').show(); }

**Comment by romanegloo** [comment]
@mkrsII I've tried your solution, but it's not updating the container list when there's no result. If the previous request had three results, it's still there when further typing results in nothing.

**Comment by mkrsII** [comment]
A better solution that i'm now using. Added a function to easy-auto to get the element count:

$.fn.getElementCount = function () {
	   
	    var inputId = $(this).attr("id");

	    if (inputId !== undefined) {
	        return EasyAutocomplete.getHandle(inputId).getElementCount();
	    }

	    return -1;
	};


this.getElementCount = function () {		    
	        return elementsList.length;
	    };

Then in the load event:

onLoadEvent: function () {
                if ($("#AutoComplete").val() !== '') {
                    var elementCount = $("#AutoComplete").getElementCount();
                    if (elementCount <= 0) {
                        $('#no-results').show();
                    }
                    else {
                        $('#no-results').hide();
                    }
                }
            }


HTML:

no-results is a DIV right below the auto complete element

**Comment by romanegloo** [comment]
@mkrsII Right, I was looking for a function like that. That should solve my problem. thanks.

**Comment by morreene** [comment]
@mkrsII  I am new to js, could you please let me know exactly where I should put your code in the file?
many thanks!

**Comment by mkrsII** [comment]
here's an example:

var options = {

  url: "resources/countries.json",

  getValue: "name",

  list: {	
    onLoadEvent: function () {
                if ($("#AutoComplete").val() !== '') {
                    var elementCount = $("#AutoComplete").getElementCount();
                    if (elementCount <= 0) {
                        $('#no-results').show();
                    }
                    else {
                        $('#no-results').hide();
                    }
                }
            }
  },


  theme: "square"
};

$("#countries").easyAutocomplete(options);

**Comment by morreene** [comment]
@mkrsII Thank you. I still have problem to get it work. here is my html code. 

<div class="easy-autocomplete" style="width: 415.828px;">
<input name="product" id="product" class="form-control" placeholder="Type something..." autocomplete="off">
  <div class="easy-autocomplete-container" id="eac-container-product">
    <ul style="display: none;">
      <li class="">
        <div class="eac-item">electric machine tools 1</div>
      </li>
....
And this is the code I used: should I replace, is it correct?


                      var options = {
                        url: "../static/hs_desc.json",
                        getValue: "label",
                        list: {
                          onLoadEvent: function () {
                                if ($("#eac-container-product").val() !== '') {
                                    var elementCount = $("#eac-container-product").getElementCount();
                                      if (elementCount <= 0) {
                                        $('#no-results').show();
                                      }
                                      else {
                                        $('#no-results').hide();
                                      }
                                    }
                                  }
                                },

                                // theme: "square"
                              };

                              $("#product").easyAutocomplete(options);

**Comment by mkrsII** [comment]
Just to make sure did you include the function in easyautocomplete.js?

$.fn.getElementCount = function () {
	   
	    var inputId = $(this).attr("id");

	    if (inputId !== undefined) {
	        return EasyAutocomplete.getHandle(inputId).getElementCount();
	    }

	    return -1;
	};


this.getElementCount = function () {		    
	        return elementsList.length;
	    };

**Comment by mkrsII** [comment]
Also, what's the error you having can you post a screen shot in the console?

**Comment by morreene** [comment]
I include the function in the HTML <scripts>. No error reported. But the search is not working, with the drop-down showing only the first six selection. 

In which part of the easy-complete.js should I put the function? Sorry for the silly question, I should learn js. :)

**Comment by mkrsII** [comment]
Difficult to see your issue. Could you put one together a sample package showing the html and js for me? Maybe use plunkr?

**Comment by morreene** [comment]
@mkrsII I have put the code in plunkr. Thanks!

https://plnkr.co/edit/ZUEZSsoSehHWY31Q4vzv?p=preview

**Comment by mkrsII** [comment]
The following must go into jquery.easy-autocomplete.min.js

$.fn.getElementCount = function () {
    	    var inputId = $(this).attr("id");
    	    if (inputId !== undefined) {
    	        return EasyAutocomplete.getHandle(inputId).getElementCount();
    	    }
    	    return -1;
    	};
    this.getElementCount = function () {
    	        return elementsList.length;
    	    };

**Comment by mkrsII** [comment]
Give me access to edit and I'll try update when I have time. Thanks.

**Comment by morreene** [comment]
Thanks, I have made the plunker public. Please when have time. Thanks!

**Comment by vishhnu-dev** [comment]
**The following code work's for me.
On  'showAnimation', remove li with class = "main-color"
On hideAnimation add li with class ="main-color" if ul.childrenl.length <=0**


showAnimation: {
										type: "slide",
										time: 400,
										callback: function() {
											$('.easy-autocomplete-container ul').next('.main-color').remove();
										}
									},
									hideAnimation: {
										type: "slide",
										time: 400,
										callback: function() {
											if( $('.easy-autocomplete-container ul').children('li').length <= 0) {
												$('.easy-autocomplete-container ul').css('display','block');
												$('.easy-autocomplete-container ul').append('<li class="main-color">Nenum resultado encontrado...</li>');
											} else {
												$('.easy-autocomplete-container ul').next('.main-color').remove();
												$('.easy-autocomplete-container ul').css('display','none');
											}
										}
									}
